src/rules/noUnsafeAnyRule.ts

```diff
@@ -47,14 +47,13 @@ export class Rule extends Lint.Rules.TypedRule {
 const isExpression: (node: ts.Node) => node is ts.Expression = (ts as any).isExpression;
 
 function walk(ctx: Lint.WalkContext<void>, checker: ts.TypeChecker): void {
-    return ts.forEachChild(ctx.sourceFile, recur);
-    function recur(node: ts.Node): void {
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
         if (isExpression(node) && isAny(checker.getTypeAtLocation(node)) && !isAllowedLocation(node, checker)) {
```

Comment:
I think it would increase performance if you swap the calls to `isAny(...)` and `isAllowedLocation(...)`